build.psm1

```diff
@@ -2023,15 +2023,48 @@ function script:precheck([string]$command, [string]$missedMessage) {
 
 # this function wraps native command Execution
 # for more information, read https://mnaoumov.wordpress.com/2015/01/11/execution-of-external-commands-in-powershell-done-right/
-function script:Start-NativeExecution([scriptblock]$sb, [switch]$IgnoreExitcode)
+function script:Start-NativeExecution
 {
+    param(
+        [scriptblock]$sb,
+        [switch]$IgnoreExitcode,
+        [switch]$VerboseOutputOnError
+    )
     $backupEAP = $script:ErrorActionPreference
     $script:ErrorActionPreference = "Continue"
     try {
-        & $sb
+        if($VerboseOutputOnError.IsPresent)
+        {
+            $output = & $sb
+        }
+        else
+        {
+            & $sb
+        }
+
         # note, if $sb doesn't have a native invocation, $LASTEXITCODE will
         # point to the obsolete value
         if ($LASTEXITCODE -ne 0 -and -not $IgnoreExitcode) {
+            if($VerboseOutputOnError.IsPresent -and $Output)
+            {
+                $output | Out-String | Write-Verbose -Verbose
+            }
+
+            # Get caller location for easier debugging
+            $callerLocation = (Get-PSCallStack -ErrorAction SilentlyContinue)[1].Location
+            if($callerLocation)
+            {
+                $callerLocationParts = $callerLocation -split ":\s*line\s*"
+                $callerFile = $callerLocationParts[0]
+                $callerLine = $callerLocationParts[1]
+
+                if ($null -ne $env:CI)
+                {
+                   Add-AppveyorCompilationMessage $errorMessage -Category Error -FileName $callerFile -Line $callerLine
```

Comment:
I don't see `$errorMessage` declared in this function. Where does it come from?

Reply:
What will show up when using `Add-AppveyorCompilationMessage`?

Reply:
It will make sure AppVeyor fails and write this message.  I fixed the message.